### ISSUE TITLE
[laserMapping-2] process has died

### ISSUE THREAD

**Issue by Cureser** [hku-mars/FAST-LIVO#143]
Linux version:Ubuntu20.04 
Opencv4.10.0

commond:roslaunch fast_livo mapping_avia_old.launch 

![download](https://github.com/user-attachments/assets/dff2fcbc-ffcb-4581-bd42-5b38a50323fc)

I don't want to install opencv3, how can i fix this problem?

**Comment by Cureser** [comment]
![_cgi-bin_mmwebwx-bin_webwxgetmsgimg__ MsgID=6542660401183095065 skey=@crypt_dd39521_99e976b00fd1423926f3437b9d3e425f mmweb_appid=wx_webfilehelper](https://github.com/user-attachments/assets/a6012e08-6465-4e97-9823-04748c6ce1a0)

This funcion appears when constructing the PinholeCamera object.

It seem that it can be found on line 1196 in the file laserMapping.cpp

**Comment by xuankuzcr** [comment]
You can directly comment out this part.

**Comment by Cureser** [comment]
Thanks!

![image](https://github.com/user-attachments/assets/dcd63cd4-d155-4ed1-b418-cf49a4f9957c)

I've commented out that part, but another process died come up.

**Comment by xuankuzcr** [comment]
Could you pull the latest code? If it's already up-to-date, could you provide some GDB information?

**Comment by Cureser** [comment]
![image](https://github.com/user-attachments/assets/f0eaad8a-eb5d-4f47-9fb1-b273ebe90201)
I‘ve commented out this part.
![image](https://github.com/user-attachments/assets/49cd6aa8-a77c-4323-a0d0-cf192097d8bb)
Here are the gdb information.
It seems that the process died during initialization.

**Comment by Cureser** [comment]
![fast-livo](https://github.com/user-attachments/assets/91d0507b-8e2b-417a-aa56-489ab83f2077)

Thanks, I successfully ran the test program by linking opencv4.2.0, but the first problem still exists, I only commented out the `cv::initUndistortRectifyMap` function so that the program would not crash.

**Comment by haoming-Yu** [comment]
Yes, I've encountered the same problem several times. Strange as it is, using opencv >= 4.0 would probably trigger this problem. I think it might have something to do with rpg_vikit's compilation and problematic dependency. Just comment out the line and then catkin_make the project again will make the program runnable and git rid of this error.
If you would like to debug this problem, just add launch-prefix="xterm -e gdb -ex run --args" to launch file's node definition, and you can find the problem of initUndistortRectifyMap function. Anyway, thanks for author's contribution and the solution for this problem. It really saved me a lot of time! :)
